### ISSUE TITLE
LammpsTrajectories not working with aiida-core 1.65

### ISSUE THREAD

**Issue by JPchico** [aiidaplugins/aiida-lammps#34]
It seems to be that it is not possible to store the zip files in the archive as it was previously done when using aiida-core 1.64 

The problem seems to be that in `create_file_from_filelike` one must pass a string instead of a bytes object, at least this is what the traceback seems to indicate when one tries to store the object.

```
aiida_lammps/tests/test_trajectory.py:90: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
aiida_lammps/data/trajectory.py:50: in __init__
    self.set_from_fileobj(handle.readlines(), aliases)
aiida_lammps/data/trajectory.py:127: in set_from_fileobj
    self.put_object_from_filelike(
/usr/share/miniconda/lib/python3.9/site-packages/aiida/orm/nodes/node.py:711: in put_object_from_filelike
    self._repository.put_object_from_filelike(handle, path, mode, encoding, force)
/usr/share/miniconda/lib/python3.9/site-packages/aiida/orm/utils/_repository.py:231: in put_object_from_filelike
    folder.create_file_from_filelike(handle, key, mode=mode, encoding=encoding)
/usr/share/miniconda/lib/python3.9/site-packages/aiida/common/folders.py:228: in create_file_from_filelike
    shutil.copyfileobj(filelike, handle)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

fsrc = <tempfile._TemporaryFileWrapper object at 0x7f11e2252550>
fdst = <_io.TextIOWrapper name='/tmp/tmpui8y2zpd/test_repo/sandbox/tmpkqa5ijui/path/trajectory.zip' mode='w' encoding='utf8'>
length = 65536

    def copyfileobj(fsrc, fdst, length=0):
        """copy data from file-like object fsrc to file-like object fdst"""
        # Localize variable access to minimize overhead.
        if not length:
            length = COPY_BUFSIZE
        fsrc_read = fsrc.read
        fdst_write = fdst.write
        while True:
            buf = fsrc_read(length)
            if not buf:
                break
>           fdst_write(buf)
E           TypeError: write() argument must be str, not bytes

/usr/share/miniconda/lib/python3.9/shutil.py:208: TypeError
```

**Comment by chrisjsewell** [comment]
Heya, I'm a bit confused by your error message here since it does not seem to relate to the code in the aiida-core `develop` branch (i.e. what will be 2.x)

Anyhow, firstly this should work with binary files now, but it is of note that it is maybe no longer necessary for aiida-lammps to handle the compression of the trajectory file, since aiida-core can handle that:

basically now in aiida 2.x, initially / by default these files are stored as individual files, e.g. it will just copy the file into the repository. But then, at a later point, the user can call `verdi storage maintain`, at which point all files be compressed into large binary files

**Comment by JPchico** [comment]
Hi @chrisjsewell 

You are absolutely right, the issue should not be 2.x it seems to be something that is different in 1.65. I have to check more carefully what is exactly causing the error. ( I have updated the issue to reflect this)
Something I noticed is that if I use the class as it was written one cannot set or parse the object. I managed to solve the parsing issue, by modifying it in my tests so that one instead reads (`readines`) the contents and they are then passed to the parser. Of this way one can parse it properly, but the same storage error keeps happening.

I also agree on your point that with the new archive it might not be necessary to have the current implementation of the trajectory file. I think some of the functionality is neat but perhaps unnecessary. I think it makes more sense to either use the "normal" trajectory or have another dedicated class, since there are several atom and time dependent quantities which can now be parsed, which one perhaps needs to see how to store in a clever way.

**Comment by JPchico** [comment]
So this seems to have been a weird issue with `git fetch upstream` not syncing something properly, as the change in the `put_object_from_filelike` call in the trajectory was not present in my fork. Adding this change everything is now solved.

**Comment by JPchico** [comment]
Fixed via 36367d08b2dfc49145ac70856744128d3c458d1a. As this is no longer a problem in aiida-2.x versions.
